sw/airborne/modules/ctrl/optical_flow_landing.c

```diff
@@ -112,56 +133,80 @@ PRINT_CONFIG_VAR(OPTICAL_FLOW_LANDING_OPTICAL_FLOW_ID)
 #define OPTICAL_FLOW_LANDING_COV_METHOD 0
 #endif
 
+// number of time steps used for calculating the covariance (oscillations)
+#ifndef OPTICAL_FLOW_LANDING_COV_WINDOW_SIZE
+#define OPTICAL_FLOW_LANDING_COV_WINDOW_SIZE 30
+#endif
+
 static abi_event agl_ev; ///< The altitude ABI event
 static abi_event optical_flow_ev;
 
 /// Callback function of the ground altitude
 static void vertical_ctrl_agl_cb(uint8_t sender_id __attribute__((unused)), float distance);
 // Callback function of the optical flow estimate:
-static void vertical_ctrl_optical_flow_cb(uint8_t sender_id __attribute__((unused)), uint32_t stamp, int16_t flow_x, int16_t flow_y, int16_t flow_der_x, int16_t flow_der_y, float quality, float size_divergence, float dist);
+static void vertical_ctrl_optical_flow_cb(uint8_t sender_id __attribute__((unused)), uint32_t stamp, int16_t flow_x,
+    int16_t flow_y, int16_t flow_der_x, int16_t flow_der_y, float quality, float size_divergence, float dist);
 
+// struct containing most relevant parameters
 struct OpticalFlowLanding of_landing_ctrl;
 
 void vertical_ctrl_module_init(void);
 void vertical_ctrl_module_run(bool in_flight);
 
+// for exponential gain landing, gain increase per second during the first (hover) phase:
+#define INCREASE_GAIN_PER_SECOND 0.02
+
 /**
  * Initialize the optical flow landing module
  */
 void vertical_ctrl_module_init(void)
 {
   unsigned int i;
 
+  // filling the of_landing_ctrl struct with default values:
   of_landing_ctrl.agl = 0.0f;
   of_landing_ctrl.agl_lp = 0.0f;
   of_landing_ctrl.vel = 0.0f;
-  of_landing_ctrl.divergence_setpoint = 0.0f;
```

Comment:
make all platform dependant variables to be initialised using settings, as you do with the pid gains.